openproblems/tasks/_batch_integration/batch_integration_graph/methods/scanorama.py

```diff
@@ -15,10 +15,13 @@
 
 
 def _scanorama(adata, use_rep):
-    from scib.integration import runScanorama
+    from scib.integration import scanorama
     from scib.preprocessing import reduce_data
 
-    adata = runScanorama(adata, "batch")
+    # scanorama clears adata.layers
+    layers = adata.layers
+    adata = scanorama(adata, "batch")
+    adata.layers = layers
     reduce_data(adata, umap=False, use_rep=use_rep)
```

Comment:
can you put `reduce_data` out of the main `_scanorama` function and into only the `scanorama_full` functions, as this is not needed for embedding outputs.

Reply:
do we not need the neighbourhood graph for full?

Reply:
this function does pca+nn for full output and only nn for embedding

Reply:
adapted that now so that it only computes pca for full output. I think it still makes sense to keep the reduce_data function in there for neighborhood graph computation